src/python/ensembl/io/genomio/genbank/extract_data.py

```diff
@@ -138,16 +134,27 @@ def parse_genbank(self, gb_file):
                 if record.id in organella:
                     record.organelle = organella[record.id]
                 self.seq_records.append(record)
+                if len(self.seq_records) >= 1:
+                    self.format_write_record()
+                else:
+                    logging.warning("No records are found in gb_file")
 
-            self._write_genome_json()
-            self._write_genes_gff()
-            self._write_seq_region_json()
-            self._write_fasta_dna()
+    def format_write_record(self) -> None:
+        """
+        Generate the prepared files from genbank record
+        """
+        self._format_genome_data()
+        self._format_write_genes_gff()
+        self._format_write_seq_json()
+        self._write_fasta_dna()
 
-    def _get_organella(self, gb_file):
+    def _get_organella(self, gb_file: PathLike) -> Dict[str, str]:
         """
         Retrieve the organelle from the genbank file, using the specific GenBank object,
         because SeqIO does not support this field
+
+        Args:
+            gb_file: path to the downloaded genbank file
```

Comment:
```suggestion
            gb_file: path to the genbank file
```
We might have the file on hand already ;)